File: iosApp/iosApp/Presentation/Main/Home/HomeAction.swift
//
//  HomeAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

enum HomeAction {
    case bannerTapped(HomeBanner)
    case maidTapped(id: String)
    case cafeTapped(id: String)
    case cafeEventTapped(cafeId: String, eventId: String)
    case birthdayMaidTapped(id: String)
    case loginPromptSignInTapped
    case dismissLoginPrompt
    case loadMorePopularCasts
    case loadMoreNearbyCafes
    case communityTapped
    case communityPostTapped(postId: String)
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/home/HomeAction.kt
package com.hhp227.concafe.presentation.main.home

import com.hhp227.concafe.domain.model.HomeBanner

sealed interface HomeAction {
    data class ClickBanner(val banner: HomeBanner) : HomeAction
    data class ClickMaid(val id: String) : HomeAction
    data class ClickCafe(val id: String) : HomeAction
    data class ClickCafeEvent(val cafeId: String, val eventId: String) : HomeAction
    data class ClickBirthdayMaid(val id: String) : HomeAction
    data object ClickLoginPromptSignIn : HomeAction
    data object DismissLoginPrompt : HomeAction
    data object LoadMoreCafeEvents : HomeAction
    data object LoadMorePopularCasts : HomeAction
    data object LoadMoreNearbyCafes : HomeAction
    data object ClickCommunity : HomeAction
    data class ClickCommunityPost(val postId: String) : HomeAction
}
